pkg/util/filesystem/util_windows_test.go

```diff
@@ -89,3 +89,12 @@ func TestPendingUnixDomainSocket(t *testing.T) {
 	wg.Wait()
 	unixln.Close()
 }
+
+func TestAbsWithSlash(t *testing.T) {
+	// On Windows, filepath.IsAbs will not return True for paths prefixed with a slash
+	assert.True(t, IsAbs("/test"))
+	assert.True(t, IsAbs("\\test"))
```

Comment:
Is a UNC path really absolute? If we're narrowly trying to make absolute unix paths pass validation, should this allow '/…' but not '//…'?

Reply:
UNC paths are absolute. It is also treated as such in golang's ``filepath.IsAbs`` [1].

Additionally, if we look at [2], we see:


... A file name is relative to the current directory if it does not begin with one of the following:
- A UNC name of any format, which always start with two backslash characters ("\\"). For more information, see the next section.
- A disk designator with a backslash, for example "C:\" or "d:\".
- A single backslash, for example, "\directory" or "\file.txt". This is also referred to as an absolute path.

According to this, UNCs are not relative.

[1] https://cs.opensource.google/go/go/+/refs/tags/go1.22.2:src/path/filepath/path_windows.go;l=64
[2] https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#fully-qualified-vs-relative-paths

Reply:
@liggitt is this info enough to resolve this conversation?

Reply:
yeah, if filepath.IsAbs returns true for UNC paths, that's fine

Reply:
I've run this on Windows:

```
package main

import (
        "fmt"
        "path/filepath"
)

func main() {
        cases := []string {
                `C:/var/lib/something`,
                `//./var/lib/something`,
                `//?/var/lib/something`,
                `//foo/C$/var/lib/something`,

                `C:\var\lib\something`,
                `\\.\var\lib\something`,
                `\\?\var\lib\something`,
                `\\foo\C$\var\lib\something`,
        }

        for _, path := range cases {
                fmt.Printf("IsAbs returns %t for: %s\n", filepath.IsAbs(path), path)
        }
}
```

Output:

```
PS C:\Users\Administrator\workdir\sys> go run .\tester.go
IsAbs returns true for: C:/var/lib/something
IsAbs returns true for: //./var/lib/something
IsAbs returns true for: //?/var/lib/something
IsAbs returns true for: //foo/C$/var/lib/something
IsAbs returns true for: C:\var\lib\something
IsAbs returns true for: \\.\var\lib\something
IsAbs returns true for: \\?\var\lib\something
IsAbs returns true for: \\foo\C$\var\lib\something
```